### DISCUSSION TITLE
[unit test]: Adding wating time before jest gives up on Strapi set-up

### DISCUSSION THREAD

**Discussion by Antoine-lb** [strapi/strapi#7549] [Legacy Topics]
Hello,
I followed the unit test set-up step by step. It kind of worked but once every three runs everything faild because of a timeout waiting for Strapi to load (5000ms by default).

Adding more time kind of fixed the issue:
```
/** this code is called once before any test is called */
beforeAll(async (done) => {
  jest.setTimeout(30000); //this line
  await setupStrapi(); // singleton so it can be called many times
  done();
});
```

I guess this will depened on everyones computer.

I know is not the perfect fix, but sometimes I can't even run the tests, it always fails (maybe becuase there is other stuff running on my computer so it makes it slow).

If you allow me I would like to put a note in the doc about this, leaving it to each person how much timeout to put. Again, I know is not ideal, but is better that than just non working in slower computers.

I will wait for your feedback.
Thanks.

**Comment by derrickmehaffy** [comment]
@Antoine-lb Please feel free to submit a pull request (and thank you for using the discussions :wink: )
